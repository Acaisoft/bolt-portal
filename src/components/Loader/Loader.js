import React from 'react'
import PropTypes from 'prop-types'
import { CircularProgress, withStyles } from '@material-ui/core'

import style from './Loader.styles'

export function Loader({ loading, children, ...rest }) {
  return loading ? <RenderLoader {...rest} /> : children
}

function RenderLoader({ fill, classes }) {
  const LoaderWithProps = function() {
    return <CircularProgress color="secondary" className={classes.root} />
  }
  return fill ? (
    <div className={classes.holder}>
      <LoaderWithProps />
    </div>
  ) : (
    <LoaderWithProps />
  )
}

Loader.propTypes = {
  loading: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  fill: PropTypes.bool,
  classes: PropTypes.object,
}

export default withStyles(style)(Loader)
