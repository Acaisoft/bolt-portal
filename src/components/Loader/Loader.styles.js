export default ({ spacing }) => ({
  root: {
    margin: spacing.unit * 2,
  },
  holder: {
    display: 'flex',
    flex: '1 1 auto',
    height: '100%',
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
})
