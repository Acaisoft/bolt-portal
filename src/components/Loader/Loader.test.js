import React from 'react'
import { shallow } from 'enzyme'

import { Loader } from './Loader'

const initLoader = overrides => {
  const mockProps = {
    children: <div id="testDiv">Hejka</div>,
    classes: {},
  }
  const wrapper = shallow(<Loader {...mockProps} {...overrides} />)
  const instance = wrapper.instance()
  return { instance, mockProps, wrapper }
}

describe('component: Loader', () => {
  describe('rendering', () => {
    it('should render without crashing', () => {
      const { wrapper } = initLoader()
      expect(wrapper).toBeTruthy()
    })

    it('should render as expected', () => {
      const { wrapper } = initLoader()
      expect(wrapper).toMatchSnapshot()
    })

    it('should render as expected when loading', () => {
      const { wrapper } = initLoader({ loading: true })
      expect(wrapper).toMatchSnapshot()
    })

    it('should render as expected when loading and when fill=true', () => {
      const { wrapper } = initLoader({ loading: true, fill: true })
      expect(wrapper).toMatchSnapshot()
    })

    it('should render loader and child when loading', () => {
      const { wrapper } = initLoader({ loading: true })
      expect(wrapper.find('RenderLoader').length).toBe(1)
      expect(wrapper.find('#testDiv').length).toBe(0)
    })

    it('should render child when not loading', () => {
      const { wrapper } = initLoader({ loading: false })
      expect(wrapper.find('CricularProgressMock').length).toBe(0)
      expect(wrapper.find('#testDiv').length).toBe(1)
    })
  })
})
