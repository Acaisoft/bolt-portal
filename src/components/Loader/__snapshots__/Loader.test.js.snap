// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Loader rendering should render as expected 1`] = `
<div
  id="testDiv"
>
  Hejka
</div>
`;

exports[`component: Loader rendering should render as expected when loading 1`] = `
<RenderLoader
  classes={Object {}}
/>
`;

exports[`component: Loader rendering should render as expected when loading and when fill=true 1`] = `
<RenderLoader
  classes={Object {}}
  fill={true}
/>
`;
